src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -100,6 +102,28 @@ public void Prepare(CancellationToken token)
                     info.NewRuleSetFilePath = newRuleSetFilePath;
                 }
             }
+
+            EnsureNoConflictingAdditionalFile();
+        }
+
+        private void EnsureNoConflictingAdditionalFile()
+        {
+            var projectSystem = serviceProvider.GetService<IProjectSystemHelper>();
+            projectSystem.AssertLocalServiceIsNotNull();
+
+            // If the correct file is already in the project, it means we were successful in adding it and there is no clash
+            if (!projectSystem.IsFileInProject(initializedProject, cSharpVBBindingConfig.AdditionalFile.Path))
+            {
+                var additionalFileName = Path.GetFileName(cSharpVBBindingConfig.AdditionalFile.Path);
+
+                var hasConflict = additionalFileConflictChecker.HasConflictingAdditionalFile(initializedProject, additionalFileName, out var conflictingAdditionalFilePath);
```

Comment:
An alternative way to write the conflict checker would have been to make it responsible for throwing exception e.g. `ThrowIfConflictingSonarLintXmlExists`. That way the checker could have thrown specific error messages for different conditions.

Just an observation - don't change it.

Reply:
Good idea, could've avoided the `out` param